targets/unity-v2/make.js

```diff
@@ -378,24 +417,34 @@ function getAuthParams(apiCall) {
     return "AuthType.None";
 }
 
-function getRequestActions(tabbing, apiCall) {
-    if (apiCall.name === "GetEntityToken")
+function getRequestActions(tabbing, apiCall, isApiInstance = false) {
+    if (apiCall.name === "GetEntityToken" && isApiInstance === false)
         return tabbing + "AuthType authType = AuthType.None;\n" +
             "#if !DISABLE_PLAYFABCLIENT_API\n" +
             tabbing + "if (authType == AuthType.None && PlayFabClientAPI.IsClientLoggedIn())\n" +
             tabbing + "    authType = AuthType.LoginSession;\n" +
             "#endif\n" +
             "#if ENABLE_PLAYFABSERVER_API || ENABLE_PLAYFABADMIN_API || UNITY_EDITOR\n" +
-            tabbing + "if (authType == AuthType.None && !string.IsNullOrEmpty(PlayFabSettings.DeveloperSecretKey))\n" +
+            tabbing + "if (authType == AuthType.None && !string.IsNullOrEmpty(request.AuthenticationContext.DeveloperSecretKey ?? PlayFabSettings.DeveloperSecretKey))\n" +
             tabbing + "    authType = AuthType.DevSecretKey;\n" +
             "#endif\n";
+	if (apiCall.name === "GetEntityToken" && isApiInstance === true)
+        return tabbing + "AuthType authType = AuthType.None;\n" +
+            "#if !DISABLE_PLAYFABCLIENT_API\n" +
+            tabbing + "if (authType == AuthType.None && authenticationContext.IsClientLoggedIn())\n" +
```

Comment:
>authenticationContext. [](start = 56, length = 22)

Please remove "authenticationContext." prefix here and call just the method IsClientLoggedIn on this API instance.

Reply:
All api instances has no IsClientLoggedInMethod for example: PlayFabAuthenticationInstanceAPI same question with above one

Reply:
Ah, i see. Ok, let's do what you did with developerSecretKey below, something like:

!string.IsNullOrEmpty(request.AuthenticationContext.ClientSessionKey ?? (authenticationContext.ClientSessionKey ?? PlayFabSettings.ClientSessionKey)

---
In reply to: [262696786](https://github.com/PlayFab/SDKGenerator/pull/420#discussion_r262696786) [](ancestors = 262696786)